sdk/tools.go

```diff
@@ -0,0 +1,9 @@
+//go:build tools
+
+package sdk
+
+import (
+	_ "go.uber.org/mock/mockgen"
+)
+
+//go:generate go run go.uber.org/mock/mockgen -package mocks -source component/component.go -destination component/internal/mocks/component_mock.go
```

Comment:
Here we're using the vendored mockgen package to generate the mocks. This doesn't require us installing mockgen locally, which can be a pain in case versions differ from machine to machine, which diffs being polluted.